crates/eframe/src/native/glow_integration.rs

```diff
@@ -1242,19 +1258,47 @@ impl GlutinWindowContext {
                     is_viewport_focused,
                     &mut viewport.screenshot_requested,
                 );
+
+                // For Wayland : https://github.com/emilk/egui/issues/4196
+                if cfg!(target_os = "linux") {
+                    let inner_size = window.inner_size();
+                    if inner_size != save_inner_size {
+                        Self::resize_for_other_os(self, viewport_id, inner_size);
+                    }
+                }
```

Comment:
Why do we need this workaround? Can we fix this bug in `winit` instead of making `eframe` more complicated?

Reply:
It's a intended and documented behaviour of winit.
In the case that the resize is handled synchronously the new size is returned from `request_inner_size` and no `Resized` event is generated.